satpy/tests/writer_tests/cf_tests/test_attrs.py

```diff
@@ -0,0 +1,144 @@
+                 'raw_metadata': OrderedDict([
+                     ('recarray', np.zeros(3, dtype=[('x', 'i4'), ('y', 'u1')])),
+                     ('flag', np.bool_(True)),
+                     ('dict', OrderedDict([('a', 1), ('b', np.array([1, 2, 3], dtype='uint8'))]))
+                 ])}
+        encoded = {'name': 'IR_108',
+                   'start_time': '2018-01-01 00:00:00',
+                   'end_time': '2018-01-01 00:15:00',
+                   'int': 1,
+                   'float': 1.0,
+                   'numpy_int': np.uint8(1),
+                   'numpy_float': np.float32(1),
+                   'numpy_bool': 'true',
+                   'numpy_void': '[]',
+                   'numpy_bytes': 'test',
+                   'numpy_string': 'test',
+                   'list': [1, 2, np.float64(3)],
+                   'nested_list': '["1", ["2", [3]]]',
+                   'bool': 'true',
+                   'array': np.array([1, 2, 3], dtype='uint8'),
+                   'array_bool': ['true', 'false', 'true'],
+                   'array_2d': '[[1, 2], [3, 4]]',
+                   'array_3d': '[[[1, 2], [3, 4]], [[1, 2], [3, 4]]]',
+                   'dict': '{"a": 1, "b": 2}',
+                   'nested_dict': '{"l1": {"l2": {"l3": [1, 2, 3]}}}',
+                   'raw_metadata': '{"recarray": [[0, 0], [0, 0], [0, 0]], '
+                                   '"flag": "true", "dict": {"a": 1, "b": [1, 2, 3]}}'}
+        encoded_flat = {'name': 'IR_108',
+                        'start_time': '2018-01-01 00:00:00',
+                        'end_time': '2018-01-01 00:15:00',
+                        'int': 1,
+                        'float': 1.0,
+                        'numpy_int': np.uint8(1),
+                        'numpy_float': np.float32(1),
+                        'numpy_bool': 'true',
+                        'numpy_void': '[]',
+                        'numpy_bytes': 'test',
+                        'numpy_string': 'test',
+                        'list': [1, 2, np.float64(3)],
+                        'nested_list': '["1", ["2", [3]]]',
+                        'bool': 'true',
+                        'array': np.array([1, 2, 3], dtype='uint8'),
+                        'array_bool': ['true', 'false', 'true'],
+                        'array_2d': '[[1, 2], [3, 4]]',
+                        'array_3d': '[[[1, 2], [3, 4]], [[1, 2], [3, 4]]]',
+                        'dict_a': 1,
+                        'dict_b': 2,
+                        'nested_dict_l1_l2_l3': np.array([1, 2, 3], dtype='uint8'),
+                        'raw_metadata_recarray': '[[0, 0], [0, 0], [0, 0]]',
+                        'raw_metadata_flag': 'true',
+                        'raw_metadata_dict_a': 1,
+                        'raw_metadata_dict_b': np.array([1, 2, 3], dtype='uint8')}
+        return attrs, encoded, encoded_flat
+
+    def assertDictWithArraysEqual(self, d1, d2):
```

Comment:
Do we add this as a util function to be imported to compare attribute dictionaries??

Reply:
Yeah this could be extracted. Not sure it needs to be a util for non-CF tests though.